### ISSUE TITLE
Admin Console Feedback - clients/credentials

### ISSUE THREAD

**Issue by stianst** [keycloak/keycloak-community#95]
https://www.keycloak.org/keycloak-community/design/admin-console/#/clients/credentials

**Comment by M3lkior** [comment]
It will be good to have  ´copy to clipboard’ buttons on « pastable » content like generated key / access token / etc

**Comment by xianli123** [comment]
> @M3lkior
> It will be good to have  ´copy to clipboard’ buttons on « pastable » content like generated key / access token / etc

Good idea. We will rethink about what can be copied and how to add the "copy to clipboard" feature.

**Comment by mposolda** [comment]
Hi, Regarding the "Signed JWT" authentication, it will be good to take the https://issues.redhat.com/browse/KEYCLOAK-10462 JIRA into account. In other words, it may be good to have tab like "Keys" for OIDC clients similarly like is available for SAML client.  In this tab, users will be able to see (or configure) JWKS URL and see (or configure) keys for various purposes (encryption, client authentication etc). So when "Signed JWT" is used as the authentication method, there may not be any configuration options (because the keys and JWKS URL will be configured at the "Keys" tab instead of here).

**Comment by xianli123** [comment]
> @mposolda
> Hi, Regarding the "Signed JWT" authentication, it will be good to take the https://issues.redhat.com/browse/KEYCLOAK-10462 JIRA into account. In other words, it may be good to have tab like "Keys" for OIDC clients similarly like is available for SAML client.  In this tab, users will be able to see (or configure) JWKS URL and see (or configure) keys for various purposes (encryption, client authentication etc). So when "Signed JWT" is used as the authentication method, there may not be any configuration options (because the keys and JWKS URL will be configured at the "Keys" tab instead of here).

Awesome. Thanks for your suggestions. We will discuss this part and think about how to design the "Keys" tab and "JWKS URL" authenticator.

**Comment by Nikhiljbhandari** [comment]
When will this new UI will be made available...with which version?

**Comment by ssilvert** [comment]
@Nikhiljbhandari, it's going to be quite some time.  We are only in the screen design phase right now.

**Comment by thomasdarimont** [comment]
I think the secrets like "client secret" etc. should be masked / hidden and shown directly with a "reveal" button and a "copy to clipboard" feature.
Otherwise secrets could be leaked when giving presentations or just reviewing client configurations.

**Comment by xianli123** [comment]
> @thomasdarimont
> I think the secrets like "client secret" etc. should be masked / hidden and shown directly with a "reveal" button and a "copy to clipboard" feature.
> Otherwise secrets could be leaked when giving presentations or just reviewing client configurations.

It makes sense. Secret and registration tokens are very important. Once it is leaked, there will be more serious consequences. We will reconsider this part. Thank you.
